lib/constructs/cost-api-implementation.ts

```diff
@@ -0,0 +1,73 @@
+import { Construct } from "constructs";
+import * as lambdaNodeJs from "aws-cdk-lib/aws-lambda-nodejs";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { Duration } from "aws-cdk-lib";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import { AtatQueue } from "./sqs";
+import { AtatRestApi } from "./apigateway";
+import { HttpMethod } from "../http";
+import * as apigw from "aws-cdk-lib/aws-apigateway";
+import { IApiRoute } from "./api-route";
+import { IResource, Method } from "aws-cdk-lib/aws-apigateway";
+
+export interface CostResourcesProps {
+  readonly environmentName: string;
+  readonly vpc?: ec2.IVpc;
+  readonly api: AtatRestApi;
+}
+
+export interface ICostApiImplementation extends IApiRoute {
+  readonly costRequestQueue: AtatQueue;
+  readonly costResponseQueue: AtatQueue;
+  readonly startCostJobFn: lambdaNodeJs.NodejsFunction;
+  readonly consumeCostResponseFn: lambdaNodeJs.NodejsFunction;
```

Comment:
So we care that these are `NodejsFunction`s? 
```suggestion
  readonly startCostJobFn: lambda.IFunction;
  readonly consumeCostResponseFn: lambda.IFunction;
```